src/jit/flowgraph.cpp

```diff
@@ -25718,3 +25720,340 @@ bool Compiler::fgUseThrowHelperBlocks()
 {
     return !opts.compDbgCode;
 }
+
+//------------------------------------------------------------------------
+// fgTailMergeThrows: Tail merge throw blocks and blocks with no return calls.
+//
+// Notes:
+//    Scans the flow graph for throw blocks and blocks with no return calls
+//    that can be merged, and opportunistically merges them.
+//
+//    Does not handle throws yet as the analysis is more costly and less
+//    likely to pay off. So analysis is restricted to blocks with just one
+//    statement.
+//
+//    For throw helper call merging, we are looking for examples like
+//    the below. Here BB17 and BB21 have identical trees that call noreturn
+//    methods, so we can modify BB16 to branch to BB21 and delete BB17.
+//
+//    Also note as a quirk of how we model flow that both BB17 and BB21
+//    have successor blocks. We don't turn these into true throw blocks
+//    until morph.
```

Comment:
Yes.

Note we only ever see this case when optimizing (since recognition of noreturn calls currently requires a failed inline attempt) and any blocks made unreachable during morph get cleaned up pretty quickly.